src/components/Search/index.jsx

```diff
@@ -0,0 +1,87 @@
+import React, { PureComponent } from 'react';
+import classNames from 'classnames';
+import PropTypes from 'proptypes';
+import './styles.scss';
+
+/* eslint-disable no-restricted-globals */
+class Search extends PureComponent {
+  constructor() {
+    super();
+    this.state = {
+      searchOptionsVisible: false,
+      searchValue: '',
+      searchCriteria: '',
+    };
+  }
+
+  componentDidUpdate() {
+    const { searchValue, searchCriteria } = this.state;
+    const { handleSearch } = this.props;
+    if (searchValue || searchCriteria) {
+      handleSearch(searchValue, searchCriteria);
+    }
+  }
+
+  toggleVisibility = () => {
+    this.setState(prevState => ({
+      searchOptionsVisible: !prevState.searchOptionsVisible,
+    }));
+  };
+
+  handleSearchCriteriaChange = (optionId) => {
+    this.setState({ searchCriteria: optionId });
+  };
+
+  handleSearchValueChange = (event) => {
+    this.setState({ searchValue: event.target.value });
+  }
+
+  handleClickOutside() {
+    this.setState({
+      searchOptionsVisible: false,
+    });
+  }
+
+  render() {
+    const { searchValue } = this.state;
+    const { options } = this.props;
+    const { title } = this.props;
+    const { searchOptionsVisible } = this.state;
+    return (
+      <div>
+        <input type="text" className="search-input" value={searchValue} onChange={this.handleSearchValueChange} />
+        <div className="search" onClick={this.toggleVisibility}>
+          <div className="search-title">
+            <span className="title">{title}</span>
+          </div>
+          {searchOptionsVisible ? (
+            <i className={classNames('fa fa-angle-down')} />
```

Comment:
I don't think we need `classNames` to combine these class names since they are not dynamically created.